packages/gateway/src/features/user/root-resolvers.ts

```diff
@@ -267,8 +271,24 @@ export const resolvers: GQLResolver = {
       })
 
       if (res.status === 403) {
+        const errorResponse = await res.json()
+        const duplicateDataErrorMap = {
+          emailForNotification: {
+            errorPrefix: 'DUPLICATE_EMAIL',
+            conflictingValue: userPayload.emailForNotification
+          },
+          mobile: {
+            errorPrefix: 'DUPLICATE_MOBILE',
+            conflictingValue: userPayload.mobile
+          }
+        }
+
+        const dupplicateError =
+          duplicateDataErrorMap[errorResponse['errorThrowingProperty']]
         return await Promise.reject(
-          new Error(`DUPLICATE_MOBILE-${userPayload.mobile}`)
+          new Error(
+            `${dupplicateError.errorPrefix}-${dupplicateError.conflictingValue}`
+          )
```

Comment:
I'm wondering if we could directly return JSON/object here so that we wouldn't need to `.split('-')[1]` when toasting about the error?